server2/Arcadia.Assistant/Arcadia.Assistant.SickLeaves/SickLeaveModelConverter.cs

```diff
@@ -6,12 +6,15 @@
 
     using Contracts;
 
+    using Employees.Contracts;
+
     public class SickLeaveModelConverter
     {
         public Expression<Func<CSP.Model.SickLeave, SickLeaveDescription>> ToDescription { get; }
             = model => new SickLeaveDescription()
             {
                 SickLeaveId = model.Id,
+                EmployeeId = new EmployeeId(model.EmployeeId),
                 StartDate = model.Start,
                 EndDate = model.End,
                 Status = model.SickLeaveCancellations.Any() ? SickLeaveStatus.Cancelled
```

Comment:
@santee, are you sure the logic is correct? In Akka version it was a bit different.

Reply:
In previous version we used timestamp of each sick leave status change to determine the most actual status:
```cs
        private CalendarEventWithAdditionalData CreateCalendarEventFromSickLeave(SickLeaves sickLeave)
        {
            var completed = sickLeave.SickLeaveCompletes
                .Select(c => new CalendarEventWithAdditionalData.SickLeaveCompletion(c.ById.ToString(), c.At))
                .FirstOrDefault();

            var cancelled = sickLeave.SickLeaveCancellations
                .Select(c => new CalendarEventWithAdditionalData.SickLeaveCancellation(c.ById.ToString(), c.At))
                .FirstOrDefault();

            var statuses = new[]
            {
                Tuple.Create(completed?.Timestamp, SickLeaveStatuses.Completed),
                Tuple.Create(cancelled?.Timestamp, SickLeaveStatuses.Cancelled)
            };

            var status = statuses
                .Where(x => x.Item1 != null)
                .OrderByDescending(x => x.Item1)
                .FirstOrDefault()
                ?.Item2;
            status = status ?? SickLeaveStatuses.Requested;

            var calendarEvent = new CalendarEvent(
                this.calendarEventIdParser.GetCalendarEventIdFromCspId(sickLeave.Id, CalendarEventTypes.Sickleave),
                CalendarEventTypes.Sickleave,
                new DatesPeriod(sickLeave.Start.Date, sickLeave.End.Date),
                status,
                sickLeave.EmployeeId.ToString());

            return new CalendarEventWithAdditionalData(calendarEvent, cancelled, completed);
        }
```

Reply:
@santee? You haven't answered yet :)